docs/pages/access-controls/login-rules/reference.mdx

```diff
@@ -0,0 +1,571 @@
+found in each element of `input` will be replaced with `replacement`.
+The matching is literal and does not support regular expressions.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`input` | `set` | set of input strings where replacements are needed |
+`match` | `string` | literal substring to be replaced |
+`replacement` | `string` | literal string to replace all instances of `match` |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`set`  | a copy of `input` where each instance of `match` found in each element of `input` has been replaced with `replacement`.
+
+#### Examples
+
+Expression | Result
+------- | ------
+`strings.replaceall(set("user-name"), "-", "_")` | `("user_name")`
+`strings.replaceall(set("user-alice", "user-bob"), "user-", "")` | `("alice", "bob")`
+
+### `ifelse`
+
+#### Signature
+
+```go
+func ifelse(cond bool, valueIfTrue any, valueIfFalse any) any
+```
+
+#### Description
+
+`ifelse` implements the classic if-else branch in a pure functional style.
+If the first argument evaluates to `true` the second argument is returned, else
+the third argument is returned.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`cond` | `bool` | condition that determines which of the following two arguments is returned |
+`valueIfTrue` | `any` | value to return if `cond` is true |
+`valueIfFalse` | `any` | value to return if `cond` is false |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`any`  | the second or third argument will be returned identically
+
+#### Examples
+
+Expression | Result
+------- | ------
+`ifelse(set("a", "b").contains("a"), set("x", "y"), set("z"))` | `("x", "y")`
+`ifelse(set("a", "b").contains("c"), set("x", "y"), set("z"))` | `("z")`
+
+### `choose`
+
+#### Signature
+
+```go
+func choose(options ...option) any
+```
+
+#### Description
+
+`choose` implements a functional style of switch statement, returning the first
+[`option`](#option) argument with a condition evaluating to `true`.
+
+If no option can be selected at runtime, this will be an error and the login
+will not succeed.
+It is recommended to add a final option with the condition hardcoded to `true`
+to implement a default option and avoid this scenario.
+For example `choose(..., option(true, set()))` would set the output to the empty
+set if no other option can be selected.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`options` | `...option` | 1 or more `option`s |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`any`  | the value of the first `option` argument with a condition evaluating to `true`
+
+#### Examples
+
+Expression | Result
+------- | ------
+`choose(option(false, set("x")), option(true, set("y")), option(true, set("z")))` | `("y")`
+`choose(option(set("a", "b").contains("a"), set("x")), option(true, set("y")))` | `("x")`
+
+### `option`
```

Comment:
Should we promote this to an H2 and organize the section like we do for `dict` and `set`? It looks like this has a constructor and a type, so it would be consistent with the other type-specific sections in the page.